enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -3427,16 +3445,23 @@ class AdjointGenerator
         Value *op1 = gutils->getNewFromOriginal(orig_ops[1]);
         Value *cmp = Builder2.CreateFCmpOLT(op0, op1);
 
+        Type *opType0 = orig_ops[0]->getType();
+        Type *opType1 = orig_ops[1]->getType();
+
         Value *diffe0 = gutils->isConstantValue(orig_ops[0])
-                            ? ConstantFP::get(orig_ops[0]->getType(), 0)
+                            ? Constant::getNullValue(opType0)
                             : diffe(orig_ops[0], Builder2);
         Value *diffe1 = gutils->isConstantValue(orig_ops[1])
-                            ? ConstantFP::get(orig_ops[1]->getType(), 0)
+                            ? Constant::getNullValue(opType1)
                             : diffe(orig_ops[1], Builder2);
 
-        Value *dif = Builder2.CreateSelect(cmp, diffe0, diffe1);
-        setDiffe(&I, dif, Builder2);
+        auto rule = [&](Value *diffe0, Value *diffe1) {
+          return Builder2.CreateSelect(cmp, diffe0, diffe1);
+        };
 
+        Value *dif =
```

Comment:
Does this work still if only one of diffe0 is vectorized/active, and the other is not?

Reply:
I would've expected for ease, you to use the diffe0 = getShadowType always, so then both are vectorized

Reply:
fixed